openff/interchange/interop/gromacs/models/models.py

```diff
@@ -341,8 +341,8 @@ class GROMACSSystem(_BaseModel):
         dict(),
         description="Molecule types, keyed by name.",
     )
-    molecules: dict[str, int] = Field(
-        dict(),
+    molecules: list[tuple[str, int]] = Field(
```

Comment:
(blocking/important) This appears to be in the public API so you might consider listing it as a behavior change in the release history.